src/linker/Linker/LinkContext.cs

```diff
@@ -286,22 +286,21 @@ public TypeDefinition GetType (string fullName)
 
 		public AssemblyDefinition TryResolve (string name)
 		{
-			try {
-				return Resolve (new AssemblyNameReference (name, new Version ()));
-			} catch (AssemblyResolutionException) {
-				return null;
-			}
+			return _resolver.Resolve (new AssemblyNameReference (name, new Version ()), _readerParameters);
 		}
 
 		public AssemblyDefinition Resolve (IMetadataScope scope)
 		{
 			AssemblyNameReference reference = GetReference (scope);
-			try {
-				AssemblyDefinition assembly = _resolver.Resolve (reference, _readerParameters);
-				return assembly;
-			} catch (Exception e) when (!(e is AssemblyResolutionException)) {
-				throw new AssemblyResolutionException (reference, e);
```

Comment:
Is the exception actually used anywhere after this change? I can't find any place which would throw it.

Reply:
This is Cecil exception which Cecil throw but not from the code path linker uses